### PR TITLE
feat: add `dynamic-plugins-info` backend plugin

### LINKED ISSUE #628: Backend Plugin that returns the list of dynamic plugin installed in the Showcase application

### PR DESCRIPTION
**Pull request by Zaperex**
## Description
Adds the `dynamic-plugins-info` backend plugin that provides the currently installed dynamic plugins in the backstage instance.

## Which issue(s) does this PR fix

- Fixes #628

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [x] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [x] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

## How to test changes / Special notes to the reviewer
Please let me know if there's a better name for this plugin, the `dynamic-plugins-info` is just a placeholder name for now.

### CODE REVIEW

File: plugins/dynamic-plugins-info-backend/src/service/router.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  const plugins = pluginManager.plugins;
+  const dynamicPlugins = plugins.map(p => {
+    // Remove the installer details for the dynamic backend plugins
+    if (p.platform === 'node') {
+      const { installer, ...rest } = p;
+      return rest as BaseDynamicPlugin;
+    }
+    return p as BaseDynamicPlugin;
+  });
+  router.get('/dynamic-plugins', (_, response) => {
```
Comment (davidfestal) [inline plugins/dynamic-plugins-info-backend/src/service/router.ts:32]: Could we name this endpoint: `loaded-plugins` ?
That would leave room for other endpoints in the future like: the new plugins that should be installed / removed after the restart, the failure to install some plugins, etc ...

Comment (Zaperex) [reply plugins/dynamic-plugins-info-backend/src/service/router.ts:32]: sure

**Comment by davidfestal** [comment]
@Zaperex  Do you think something can be done to reduce the [code duplication raised by SonarCloud](https://sonarcloud.io/component_measures?metric=new_duplicated_lines&selected=janus-idp_backstage-showcase%3Aplugins%2Fdynamic-plugins-info-backend%2F__fixtures__&view=list&pullRequest=638&id=janus-idp_backstage-showcase)  ?

**Comment by Zaperex** [comment]
Hmm it's flagging the mock data and mock result which are very similar. I guess I can try to get sonarcloud to ignore those files?

**Review by davidfestal — APPROVED** [review]

**Comment by Zaperex** [comment]
Currently having issues creating a merge commit due to `lint-staged` failing due to it trying to lint everything including the dynamic plugins wrappers, then starts complaining about missing `tsconfig.json` files? When I try excluding the `e2e-tests/**` and `dynamic-plugins/wrappers/**` directories, it then complains about the `packages/backend` missing a `tsconfig.json`
<details>
<summary>Abridged Error Logs</summary>

```bash
> yarn lint-staged
yarn run v1.22.19
$ lint-staged
✔ Preparing lint-staged...
⚠ Running tasks for staged files...
  ❯ package.json — 167 files
    ❯ * — 167 files
      ✖ turbo run prettier:fix -- [KILLED]
    ❯ *.{jsx,ts,tsx,mjs,cjs} — 56 files
      ✖ turbo run lint --concurrency=1 -- --fix [KILLED]
    ❯ !(.eslintrc).js — 1 file
      ✖ turbo run lint --concurrency=1 -- --fix [FAILED]
↓ Skipped because of errors from tasks.
✔ Reverting to original state because of errors...
✔ Cleaning up temporary files...

✖ turbo run lint --concurrency=1 -- --fix:
roadiehq-backstage-plugin-argo-cd-backend:lint: ERROR: command finished with error: command (/home/frkong/coding/janus-idp/backstage-showcase/dynamic-plugins/wrappers/roadiehq-backstage-plugin-argo-cd-backend-dynamic) yarn run lint --fix /home/frkong/coding/janus-idp/backstage-showcase/e2e-test/cypress/plugins/index.js exited (1)
roadiehq-backstage-plugin-argo-cd-backend#lint: command (/home/frkong/coding/janus-idp/backstage-showcase/dynamic-plugins/wrappers/roadiehq-backstage-plugin-argo-cd-backend-dynamic) yarn run lint --fix /home/frkong/coding/janus-idp/backstage-showcase/e2e-test/cypress/plugins/index.js exited (1)
 ERROR  run failed: command  exited (1)
• Packages in scope: @internal/plugin-dynamic-plugins-info-backend, @internal/plugin-scalprum-backend, app, backend, backstage-plugin-azure-devops-backend, backstage-plugin-catalog-backend-module-github, backstage-plugin-catalog-backend-module-github-org, backstage-plugin-catalog-backend-module-gitlab, backstage-plugin-jenkins-backend, backstage-plugin-kubernetes-backend, backstage-plugin-scaffolder-backend-module-gitlab, backstage-plugin-sonarqube-backend, backstage-plugin-techdocs-backend, dynamic-plugins-imports, dynamic-plugins-utils, immobiliarelabs-backstage-plugin-gitlab-backend, janus-idp-backstage-plugin-aap-backend, janus-idp-backstage-plugin-keycloak-backend, janus-idp-backstage-plugin-ocm-backend, roadiehq-backstage-plugin-argo-cd-backend, roadiehq-scaffolder-backend-argocd, roadiehq-scaffolder-backend-module-utils
• Running lint in 22 packages
• Remote caching disabled
roadiehq-backstage-plugin-argo-cd-backend:lint: cache miss, executing dfb189c16ec31382
$ backstage-cli package lint --fix /home/frkong/coding/janus-idp/backstage-showcase/e2e-test/cypress/plugins/index.js
roadiehq-backstage-plugin-argo-cd-backend:lint: 
roadiehq-backstage-plugin-argo-cd-backend:lint: 
roadiehq-backstage-plugin-argo-cd-backend:lint:   ✘  http://eslint.org/docs/rules/
roadiehq-backstage-plugin-argo-cd-backend:lint: 
roadiehq-backstage-plugin-argo-cd-backend:lint:      Parsing error: Cannot read file '/home/frkong/coding/janus-idp/backstage-showcase/dynamic-plugins/wrappers/roadiehq-backstage-plugin-argo-cd-backend-dynamic/tsconfig.json'
roadiehq-backstage-plugin-argo-cd-backend:lint: 
roadiehq-backstage-plugin-argo-cd-backend:lint: 
roadiehq-backstage-plugin-argo-cd-backend:lint:      e2e-test/cypress/plugins/index.js:0:0
roadiehq-backstage-plugin-argo-cd-backend:lint: 
roadiehq-backstage-plugin-argo-cd-backend:lint: ✘ 1 problem (1 error, 0 warnings)
roadiehq-backstage-plugin-argo-cd-backend:lint: 
roadiehq-backstage-plugin-argo-cd-backend:lint: 
roadiehq-backstage-plugin-argo-cd-backend:lint: Errors:
roadiehq-backstage-plugin-argo-cd-backend:lint:   1  http://eslint.org/docs/rules/null
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
</details>

**Comment by Zaperex** [comment]
For the sonarcloud code analysis, should I just remove the test data?

File: plugins/dynamic-plugins-info-backend/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+Welcome to the dynamic-plugins-info backend plugin!
+
+This plugin depends on the `backend-plugin-manager` and lists all the dynamic plugins installed in the dynamic plugins root folder.
+
+## Getting started
+
+This plugin has been added to the backend app in this repository, meaning you'll be able to access it by running `yarn
+start-backend` in the root directory, and then navigating to [/api/dynamic-plugins-info](http://localhost:7007/api/dynamic-plugins-info).
+
+To view the list of installed dynamic plugins, navigate to `http://localhost:7007/api/dynamic-plugins-info/dynamic-plugins``
```
Comment (kadel) [inline plugins/dynamic-plugins-info-backend/README.md:12]:
```suggestion
To view the list of installed dynamic plugins, navigate to `http://localhost:7007/api/dynamic-plugins-info/dynamic-plugins`
```

File: plugins/dynamic-plugins-info-backend/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+Welcome to the dynamic-plugins-info backend plugin!
+
+This plugin depends on the `backend-plugin-manager` and lists all the dynamic plugins installed in the dynamic plugins root folder.
+
+## Getting started
+
+This plugin has been added to the backend app in this repository, meaning you'll be able to access it by running `yarn
+start-backend` in the root directory, and then navigating to [/api/dynamic-plugins-info](http://localhost:7007/api/dynamic-plugins-info).
+
+To view the list of installed dynamic plugins, navigate to `http://localhost:7007/api/dynamic-plugins-info/dynamic-plugins`
```
Comment (kadel) [inline plugins/dynamic-plugins-info-backend/README.md:12]:
```suggestion
To view the list of installed dynamic plugins, navigate to `http://localhost:7007/api/dynamic-plugins-info/loaded-plugins`
```

File: yarn.lock [hunk 4]
```diff
@@ -16130,7 +16130,7 @@ express-prom-bundle@6.6.0:
     on-finished "^2.3.0"
     url-value-parser "^2.0.0"
 
-express-promise-router@^4.1.0, express-promise-router@^4.1.1:
+express-promise-router@4.1.1, express-promise-router@^4.1.0, express-promise-router@^4.1.1:
```
Comment (kadel) [inline yarn.lock:16133]: 4.1.1 is defined twice; is this ok?

Comment (Zaperex) [reply yarn.lock:16133]: Oh one is for 4.11 and the other is for ^4.1.1, so I don't think it should be an issue?

**Comment by kadel** [comment]
>Oh one is for 4.11 and the other is for ^4.1.1, so I don't think it should be an issue?

sorry, I already deleted the original comment when I noticed that one has `^` 😇

**Review by kadel — APPROVED** [review]

**Pull request merged**